utils/lpac/utils.c

```diff
@@ -7,31 +7,24 @@
 #include <stdlib.h>
 #include <string.h>
 
-static bool is_numeric(const char *value) {
-    if (value == NULL)
-        return false;
-    for (size_t i = strlen(value); i > 0; --i) {
-        if (isdigit(value[i]))
-            continue;
-        return false;
-    }
-    return true;
-}
-
 const char *getenv_str_or_default(const char *name, const char *default_value) {
     const char *value = getenv(name);
-    if (value == NULL)
+    if (value == NULL || strlen(value) == 0)
         return default_value;
     return value;
 }
 
 bool getenv_bool_or_default(const char *name, const bool default_value) {
     const char *value = getenv(name);
-    if (value == NULL)
+    if (value == NULL || strlen(value) == 0)
         return default_value;
-    if (is_numeric(value))
-        return strcmp(value, "0") != 0;
-    return strcasecmp(value, "y") == 0 || strcasecmp(value, "yes") == 0 || strcasecmp(value, "true") == 0;
+    // clang-format off
+    return strcasecmp(value, "1") == 0
+        || strcasecmp(value, "y") == 0
+        || strcasecmp(value, "on") == 0
+        || strcasecmp(value, "yes") == 0
+        || strcasecmp(value, "true") == 0;
```

Comment:
Actually, many software choose to define some special values as false, and all other non-empty values as ture.